debugger.c

```diff
@@ -12,7 +12,22 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <inttypes.h>
+
+#if __APPLE__
```

Comment:
This is repeated a lot. Can we have something like this:

    #if __APPLE__ && (TARGET_IPHONE_SIMULATOR || TARGET_OS_IPHONE)
    #include "ios.h"
    #endif

and do the magic there?

Even if different .c files want different iOS headers to be included, we could just include all of them in "ios.h"; no harm in that.

Also, why special case Mac? What's "TargetConditionals.h"?

Reply:
The mac uses the standard code, TargetConditionals.h allows you to target specific apple platforms.